synapse/handlers/auth.py

```diff
@@ -888,10 +889,10 @@ async def validate_login(
             LoginError if there was an authentication problem.
         """
 
-        if username.startswith("@"):
-            qualified_user_id = username
-        else:
-            qualified_user_id = UserID(username, self.hs.hostname).to_string()
+        # We need a fully qualified User ID for some method calls here
+        qualified_user_id = username
```

Comment:
Can we just rename the input? (Or rename all users of `qualified_user_id`?)

Reply:
`username` could be a localpart here, however we need both the original input, as well as qualified user IDs for this code, so we create two variables.

Reply:
Descriptions of all the password auth provider functions are here: https://github.com/matrix-org/synapse/blob/develop/docs/password_auth_providers.md

`check_auth` is supposed to be what the user provided, localpart or not.

Reply:
I was not clear in my comment, sorry.

I meant that I found it odd for a function to have a `username` parameter which is used in one place (to save it into a separate variable). That's why I was asking about renaming all uses of `qualified_user_id` to `username` since we only seem to use it once.

Anyway, this change is really just a style change and renaming all of that would make the diff bigger, so ignore this comment and let's leave it.